docs/documentation/server_admin/topics/admin-console-permissions/fine-grain-v2.adoc

```diff
@@ -178,60 +189,108 @@ assigning only *manage* without *view* will prevent the user from being visible.
 * *Enforce access to*: Defines if the permission should enforce access to all resources of the selected type or to specific resources in a realm.
 * *Policies*: Defines a set of one or more policies that should be evaluated to grant or deny access to the selected resource(s).
 
+After creating the permission, it will automatically take effect when enforcing access to (all) resources and scopes you selected.
+Please, keep that in mind that when creating and updating permissions in production.
+
+===== Defining permissions for viewing realm resources
+
+This feature relies on a partial evaluation mechanism to partially evaluate the permissions that a realm administrator has
+when listing and viewing realm resources. This mechanism will pre-fetch all the permissions set for view-related scopes where the realm administrator
+is referenced either directly or indirectly.
+
+Permissions that grant access to `view` a realm resource of a certain type must use one of the following policies to
+make them available from queries:
+
+* `User`
+* `Group`
+* `Role`
+
+By using any of the policies above, {project_name} can pre-calculate the set of resources that a realm administration can view
+by looking for a direct (if using a user policy) or indirect (if using a role or group policy) reference to the realm administrator.
+Therefore, decorating queries with access controls that will run at the database level. This capability is mainly important to
+properly allow paginating resources as well as avoid an additional overhead on the server-side when evaluating permissions for each
+realm resource returned by queries.
+
+Partial evaluation and filtering will only take place if the feature is enabled to a realm, and if the user is not granted
+with view-related administrative roles like `view-users` or `view-clients`. For instance, it will not happen for regular server administrators granted
+with the `admin` role.
+
+When querying resources, the partial evaluation mechanism works as follows:
+
+* Resolve all the permissions for a certain resource type that reference the realm administrator
+* Pre-evaluate each permission to check if the realm administrator is allowed or not access to the resources associated with the permission
+* Decorate database queries based on the resources granted or denied
+
+As a result, the result set of a query will hold only the realm resources where realm administrators have access to any of the view-related scopes.
```

Comment:
```suggestion
As a result, the resulting set of a queries hold only the realm resources where realm administrators have access to any of the view-related scopes.
```

Reply:
I would keep it as is because queries have a result set, the data returned from the database.

Reply:
Okay that's fine.